comp110-worksheet-6/OxoBoard.cs

```diff
@@ -7,42 +7,210 @@
+            }
+           
+           
+
+        
+
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            // If statement that checks if cords are inside the board
+            if((x >= 0 && x <= 2) && (y >= 0 && y <= 2))
+            {
+                if (boardArray[x, y] == Mark.None)
+                {
+                    boardArray[x, y] = mark;
+                    return true;
+                    // Returns true if board tile was empty and was now replaced by X or O
+                }
+                else
+                {
+                    return false;
+                    // Returns flase if tile was already occupied and leaves tile as is
+                }
+            }
+            else
+            {
+                return false;
+                // Returns false as invalid move if player inputted cords that were off the 3x3 grid
+            }
+          
+            
+
+            
+
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            int filledTiles = 0;
+            // Int set for amount of tiles that are filled
+
+            // For loops used again to check every tile on the 3x3 grid
+            for (int x = 0; x < 3; x++)
+            {
+                for (int y = 0; y < 3; y++)
+                {
+                    if(boardArray[x , y] == Mark.None)
+                    {
+                        // Board is empty on this tile
+                    }
+                    else
+                    {
+                        filledTiles++;
+                        // Board tile is occupied
+                    }
+                }
+            }
+
+            // After checking every tile if all tiles are filled filedTiles will be 9
+            if(filledTiles < 9)
+            {
+                // If there is not 9 filledTiles the board still has spaces
+                return false;
+            }
+            else
+            {
+                // Else if there is 9 filledTiles then the board is full
+                return true;
+            }
+        }
 
 		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
 		// Otherwise, return Mark.None.
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+
+            // Here I use if statements to check every possible way of winning
+            // Quite inefficent but still works for determining every winning outcome on a 3x3 grid
```

Comment:
Indeed this is quite inefficient (maybe not in terms of speed but certainly in terms of maintainability and expandability) -- would be good to try and refactor this to make it more compact e.g. making more use of loops